src/adagrams.js

```diff
@@ -1,8 +1,198 @@
+    while (i < count){
+      poolArray.push(letter);
+      i ++; 
+    }
+  }
+
+  return poolArray;
+}; 
+
+// shuffle array using Fisher Yates shuffle algorithim : https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle
+// source: https://stackoverflow.com/questions/2450954/how-to-randomize-shuffle-a-javascript-array/2450976#2450976
+const _shuffleArray = function (array) {
+	let currentIndex = array.length;
+	let temporaryValue, randomIndex;
+
+	while (0 !== currentIndex) {
+		randomIndex = Math.floor(Math.random() * currentIndex); // produce random int
+		currentIndex -= 1;
+
+		// swap in place
+		temporaryValue = array[currentIndex];
+		array[currentIndex] = array[randomIndex];
+		array[randomIndex] = temporaryValue;
+	}
+
+	return array;
+};
+
+// record the number of occurences of each element in an array
+const _elementCount = function (array) {
+  const obj = {}; 
+
+  array.forEach( function(i) {
+    obj[i] = (obj[i] || 0) + 1;
+  } ); 
+
+  return obj;
+};
+
+// ----- MAIN OBJECT -----
+
 const Adagrams = {
+  pool: {
+    A : 9, 	
+    B : 2,	
+    C : 2, 	
+    D : 4, 	
+    E : 12, 	
+    F : 2, 	
+    G : 3, 	
+    H : 2, 	
+    I : 9, 	
+    J : 1, 	
+    K : 1, 	
+    L : 4, 	
+    M : 2, 	
+    N : 6,
+    O : 8,
+    P : 2,
+    Q : 1,
+    R : 6,
+    S : 4,
+    T : 6,
+    U : 4,
+    V : 2,
+    W : 2,
+    X : 1,
+    Y : 2,
+    Z : 1
+  },
+
+  scoreChart : {
+    1 : ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+    2 : ['D', 'G'],
+    3 : ['B', 'C', 'M', 'P'],
+    4 : ['F', 'H', 'V', 'W', 'Y'],
+    5 : ['K'],
+    8 : ['J', 'X'],
+    10 : ['Q', 'Z']
+  }, 
+
   drawLetters() {
-    // Implement this method for wave 1
+    // set up data
+    const hand = [];
+    const letterArray = _createArray(this.pool);
+    _shuffleArray(letterArray);
+
+    // choose hand
+    for(let i = 0; i < 10 ; i++) {
+      hand.push(letterArray[i]);
+    }
+
+    return hand;
+  },
+
+  usesAvailableLetters(input, lettersInHand) {
+    // set up data
+    input = input.toUpperCase();
+    const inputArray = input.split('');
+    const handLetterCounts = _elementCount(lettersInHand);
+    const inputLetterCounts = _elementCount(inputArray);
+
+    // word should not be bigger than hand
+    if(inputArray.length > lettersInHand.length) return false;
+
+    // check for mismatching letters
+    for(let letter in inputArray) {
+      if (lettersInHand.includes(inputArray[letter]) === false) return false;
+    }; 
+
+    // compare count occurences    
+    for(let key in inputLetterCounts) {
+      if(inputLetterCounts[key] > handLetterCounts[key]) return false;
+    };
+  
+    return true;
+  },
+
+  scoreWord(word) {
+    word = word.toUpperCase();
+    const wordArray = word.split('');
+    let score = 0; // assume no negative scores
+
+    // add bonus points for long words
+    if(wordArray.length >= 7 && wordArray.length <= 10) score += 8 ;
+    
+    // calculate score
+    for(let letter in wordArray) {
```

Comment:
when used on arrays, for ... in loops will make `letter` the index you're looking at. If you want to look at each element in `wordArray`, consider refactoring this to a `forEach` loop! That way, below, you won't have to do `wordArray[letter]`, but just `letter`